### PR TITLE
[Driver][RISCV] Forward --no-relax option to linker for RISC-V on *BS…

### PR DESCRIPTION
**Pull request by brad0**
…D, Fuchsia and Haiku

Based on https://github.com/llvm/llvm-project/pull/76432

### CODE REVIEW

**Comment by MaskRay** [comment]
Test?

File: clang/test/Driver/freebsd.c [hunk 1]
```diff
@@ -203,3 +203,12 @@
 // RELOCATABLE-NOT: "-l
 // RELOCATABLE-NOT: crt{{[^./\\]+}}.o
 
+// Check that the -X flag is passed to the linker on riscv64
```
Comment (MaskRay) [inline clang/test/Driver/freebsd.c:206]: The CHECK is self-explanatory and renders the comment redundant. The RUN line may test additional options in the future.

**Review by MaskRay — APPROVED** [review]

File: clang/test/Driver/freebsd.c (hunk 1 above)
Comment (brad0) [reply clang/test/Driver/freebsd.c:206]: When I was putting the diff together I was questioning whether I should combine the two into one instance.

**Comment by brad0** [comment]
I fixed up the tesets.

**Pull request merged**